tests/spring-cloud-sleuth-instrumentation-reactor-tests/src/test/java/org/springframework/cloud/sleuth/instrument/web/client/WebClientBraveTests.java

```diff
@@ -128,7 +118,8 @@ public void onError(Throwable t) {
 			@Override
 			public void onComplete() {
 				if (ref.getAndSet(null) != null) {
-					callback.onSuccess(null);
+					callback.onError(
```

Comment:
isn't that path testing the "empty source" case? ie. if the `ref` wasn't set to `null` previously, it means that we completed directly, aka. the source is empty. this is not an error. note that a well-formed source MUST guarantee `onNext`-before-`onComplete` order.

Reply:
again, since that is dedicated to a specific test, having such an early completion could indeed be considered an error (ie. we expect exactly `onNext(ClientResponse)` then `onComplete()`).

Reply:
we have so few tests about these edge cases.. https://github.com/spring-cloud/spring-cloud-sleuth/issues/1570

main thing is that the two types of integration tests here, WebClient and HttpClient should have a coherent means to do the subscription dance, or a clear reason why they are different. This PR so far attempts to get them to do the same thing.. then we can start documenting all the odd possibilities therein

Reply:
empty source, we'll handle explicitly here. https://github.com/spring-cloud/spring-cloud-sleuth/issues/1570#issuecomment-591247178

thanks for the reminder. this particular one is test code which isn't written to also handle empty source, plus there's no empty source test, yet